### PR TITLE
Merge ign-gui6 ➡️  main

### PR DESCRIPTION
**Pull request by chapulina**
# ➡️  Forward port

Port `ign-gui6 ` ➡️  `main`

Branch comparision: https://github.com/ignitionrobotics/ign-gui/compare/main...ign-gui6

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

### CODE REVIEW

**Review by jennuine — APPROVED** [review]

File: src/Dialog_TEST.cc [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+#include <gtest/gtest.h>
+
+#include <gz/common/Console.hh>
+#include <gz/utils/ExtraTestMacros.hh>
+
+#include "test_config.hh"  // NOLINT(build/include)
+#include "gz/gui/Application.hh"
+#include "gz/gui/Dialog.hh"
+
+std::string kTestConfigFile = "/tmp/ign-gui-test.config"; // NOLINT(*)
```
Comment (jennuine) [inline src/Dialog_TEST.cc:27]:
```suggestion
std::string kTestConfigFile = "/tmp/gz-gui-test.config"; // NOLINT(*)
```

File: src/cmd/CMakeLists.txt [hunk 2]
```diff
@@ -11,4 +11,14 @@ configure_file(
...
+
+# Tack version onto and install the bash completion script
+configure_file(
+  "gui.bash_completion.sh"
+    "${CMAKE_CURRENT_BINARY_DIR}/gui${PROJECT_VERSION_MAJOR}.bash_completion.sh" @ONLY)
+install(
+  FILES
+    ${CMAKE_CURRENT_BINARY_DIR}/gui${PROJECT_VERSION_MAJOR}.bash_completion.sh
+  DESTINATION
+    ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATAROOTDIR}/gz/gz${IGN_TOOLS_VER}.completion.d)
```
Comment (jennuine) [inline src/cmd/CMakeLists.txt:24]:
```suggestion
    ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATAROOTDIR}/gz/gz${GZ_TOOLS_VER}.completion.d)
```

File: src/cmd/gui.bash_completion.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# bash tab-completion
+
+# This is a per-library function definition, used in conjunction with the
+# top-level entry point in ign-tools.
```
Comment (jennuine) [inline src/cmd/gui.bash_completion.sh:21]:
```suggestion
# top-level entry point in gz-tools.
```

File: src/gz_TEST.cc [hunk 4]
```diff
@@ -96,3 +101,34 @@ TEST_F(CmdLine, GZ_UTILS_TEST_ENABLED_ONLY_ON_LINUX(list))
   EXPECT_TRUE(common::exists(common::joinPaths(this->kFakeHome, ".gz",
       "gui")));
 }
+
+//////////////////////////////////////////////////
+/// \brief Check --help message and bash completion script for consistent flags
+// See https://github.com/gazebo-tooling/release-tools/issues/398
+TEST(ignTest, GZ_UTILS_TEST_DISABLED_ON_WIN32(GuiHelpVsCompletionFlags))
```
Comment (jennuine) [inline src/gz_TEST.cc:108]:
```suggestion
TEST(gzTest, GZ_UTILS_TEST_DISABLED_ON_WIN32(GuiHelpVsCompletionFlags))
```

File: tutorials/07_config.md [hunk 5]
```diff
@@ -15,9 +15,19 @@ By default, Gazebo GUI will load the config file at
...
+`gz gui -c /absolute/path/to/example.config`
+
+Or a path relative to the current working directory:
+
+`gz gui -c relative/path/to/example.config`
+
+Or a path relative to a custom directory, which you can specify by setting the
+environment variable `GZ_GUI_RESOURCE_PATH`, like so:
+
+`GZ_GUI_RESOURCE_PATH=/absolute/path/to/ ign gui --config example.config`
```
Comment (jennuine) [inline tutorials/07_config.md:30]:
```suggestion
`GZ_GUI_RESOURCE_PATH=/absolute/path/to/gz gui --config example.config`
```

File: tutorials/07_config.md [hunk 6]
```diff
@@ -15,9 +15,19 @@ By default, Gazebo GUI will load the config file at
...
 
 From the C++ API, pass the file path to
 [Application::LoadConfig](https://gazebosim.org/api/gui/6.0/classignition_1_1gui_1_1Application.html#a03c4c3a1b1e58cc4bff05658f21fff17).
```
Comment (jennuine) [inline tutorials/07_config.md:33]:
```suggestion
[Application::LoadConfig](https://gazebosim.org/api/gui/7.0/classignition_1_1gui_1_1Application.html#a03c4c3a1b1e58cc4bff05658f21fff17).
```

**Review by jennuine — COMMENTED** [review]
I found a couple more renaming places. Do we need to move `include/ignition/gui/*` to `include/gz/gui`? Also, are we renaming `IgnHelpers.qml` to `GzHelpers.qml`?

**Comment by chapulina** [comment]
> I found a couple more renaming places. Do we need to move include/ignition/gui/* to include/gz/gui? Also, are we renaming IgnHelpers.qml to GzHelpers.qml?

Nice catch, I'll look into that soon

**Comment by chapulina** [comment]
> Do we need to move include/ignition/gui/* to include/gz/gui? Also, are we renaming IgnHelpers.qml to GzHelpers.qml?

Ahh I forgot that those were duplicated. Fixed in 593ccb0680a6b8890cd0e7b8cab6421bccbcbda8. I'd like to revisit that eventually and see if we can get away without duplicated files.

File: include/gz/gui/resources.qrc [hunk 7]
```diff
@@ -39,8 +42,11 @@
     <!-- This qmldir file defines a QML module that can be imported -->
     <file alias="qmldir">qml/qmldir</file>
     <!-- Add any QML components referenced in the qmldir file here -->
+    <file alias="GzColor.qml">qml/GzColor.qml</file>
+    <file alias="GzPose.qml">qml/GzPose.qml</file>
     <file alias="GzSnackBar.qml">qml/GzSnackBar.qml</file>
     <file alias="GzSpinBox.qml">qml/GzSpinBox.qml</file>
+    <file alias="GzVector3.qml">qml/GzVector3.qml</file>
     <file alias="IgnSnackBar.qml">qml/GzSnackBar.qml</file>
     <file alias="IgnSpinBox.qml">qml/GzSpinBox.qml</file>
```
Comment (jennuine) [inline include/gz/gui/resources.qrc:51]:
```suggestion
    <file alias="GzSnackBar.qml">qml/GzSnackBar.qml</file>
    <file alias="GzSpinBox.qml">qml/GzSpinBox.qml</file>
```

**Review by jennuine — COMMENTED** [review]
Are these aliases still needed with 593ccb0680a6b8890cd0e7b8cab6421bccbcbda8?: https://github.com/gazebosim/gz-gui/blob/593ccb0680a6b8890cd0e7b8cab6421bccbcbda8/include/gz/gui/resources.qrc#L30-L39

File: include/gz/gui/resources.qrc (hunk 7 above)
Comment (chapulina) [reply include/gz/gui/resources.qrc:51]: I believe we can get rid of the duplicate QML files, let's try to get this one first:

* https://github.com/gazebosim/gz-gui/pull/449

File: src/plugins/topic_echo/TopicEcho_TEST.cc [hunk 8]
```diff
@@ -16,224 +16,264 @@
...
+#endif
+#include <gz/msgs/stringmsg.pb.h>
+#ifdef _MSC_VER
+#pragma warning(pop)
+#endif
+
+#include <gz/common/Console.hh>
+#include <gz/common/Filesystem.hh>
 #include <gz/transport/Node.hh>
+#include <gz/utilities/ExtraTestMacros.hh>
```
Comment (mayman99) [inline src/plugins/topic_echo/TopicEcho_TEST.cc:32]:
```suggestion
```

**Comment by chapulina** [comment]
I can reproduce the HFOV test failure locally, but I haven't figured out how to fix it:

```
   /github/workspace/test/integration/minimal_scene.cc:158: Failure
  The difference between 60 and camera->HFOV().Degree() is 15.009121067327513, which exceeds 1e-4, where
  60 evaluates to 60,
  camera->HFOV().Degree() evaluates to 75.009121067327513, and
  1e-4 evaluates to 0.0001.
```

It looks like the aspect ratio is changing in the middle of the test, so the HFOV changes. @AzulRadio @jennuine , any tips?

**Comment by jennuine** [comment]
Currently, I think there is a bug in `gz-rendering`. This line here seems to be the source of the issue: https://github.com/gazebosim/gz-gui/blob/6059daacf60fcadfaee952806a710342eb4f1b2e/src/plugins/minimal_scene/MinimalScene.cc#L321

Looking at `camera->HFOV()` before the `camera->PreRender()` call, the `HFOV()` returns `60` degrees. After `camera->PreRender()` it returns `75.0091` degrees, which cases the test to fail. I'll take a closer look at `gz-rendering` tomorrow

**Comment by chapulina** [comment]
> Looking at camera->HFOV() before the camera->PreRender() call, the HFOV() returns 60 degrees. After camera->PreRender() it returns 75.0091 degrees

Thanks for the lead, @jennuine ! I believe the issue is that we were setting the aspect ratio every time the texture is dirty, but now that we want to keep a constant FOV, we should be setting that instead. Also since https://github.com/gazebosim/gz-rendering/pull/635 the aspect ratio is updated automatically with the `SetImage[Height|Width]` calls.

See the fix in [c7cac24](https://github.com/gazebosim/gz-gui/pull/446/commits/c7cac24e72c8dfc3aa8e3b873c0fcaada9c8adbd) and let me know if you're ok with it. The test passes now, and I checked locally that resizing a window behaves correctly.

**Review by jennuine — APPROVED** [review]
> See the fix in [c7cac24](https://github.com/gazebosim/gz-gui/pull/446/commits/c7cac24e72c8dfc3aa8e3b873c0fcaada9c8adbd) and let me know if you're ok with it

LGTM!

**Comment by jennuine** [comment]
Are we still deprecating `gz-sim` `scene3d`? If not, we should update it too (and update the deprecated warning version numbers)l: https://github.com/gazebosim/gz-sim/blob/ff101099698da7795fe25f9c03b0bc34a7a25f5e/src/gui/plugins/scene3d/Scene3D.cc#L688-L689

**Comment by chapulina** [comment]
> Are we still deprecating gz-sim scene3d?

Yup, that's being removed in 

* https://github.com/gazebosim/gz-sim/pull/1605

**Pull request merged**